tests/test_ci_app_host_home_isolation.py

```
@@ -0,0 +1,446 @@
+        "target-bound cleanup confirmation": "CMUX_APP_HOST_CLEANUP_CONFIRMATION",
+        "external confirmation record": "CMUX_APP_HOST_CONFIRMATION_FILE",
+        "structured Ghostty config sentinel": "cmux CI app-host isolation sentinel",
+        "owner-only app-host access": 'chmod -R u+rwX,go-rwx "$app_host_home"',
+    }.items():
+        require(prepare_app_host, needle, context)
+    publish = prepare_app_host.index('>> "$GITHUB_ENV"')
+    first_home_mutation = prepare_app_host.index('mkdir -m 700 "$app_host_home"')
+    if publish > first_home_mutation:
+        raise SystemExit(
+            "FAIL: identity and cleanup target must be published before mutation"
+        )
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)